tools/get_features.py

```diff
@@ -0,0 +1,303 @@
+from tarfile import DEFAULT_FORMAT
+import fire
+import numpy as np
+import pandas as pd
+import pathlib
+from penquins import Kowalski
+import warnings
+from typing import List, Union
+import yaml
+import json
+import os
+import time
+import h5py
+
+BASE_DIR = os.path.dirname(__file__)
+JUST = 50
+TIMEOUT = 300
+
+
+config_path = pathlib.Path(__file__).parent.parent.absolute() / "config.yaml"
+with open(config_path) as config_yaml:
+    config = yaml.load(config_yaml, Loader=yaml.FullLoader)
+
+kowalski = Kowalski(
+    token=config["kowalski"]["token"],
+    protocol=config["kowalski"]["protocol"],
+    host=config["kowalski"]["host"],
+    port=config["kowalski"]["port"],
+    timeout=TIMEOUT,
+)
+
+
+def get_features(
+    source_ids: List[int],
+    features_catalog: str = "ZTF_source_features_DR5",
+    verbose: bool = False,
+    restart: bool = True,
+    **kwargs,
+):
+
+    ccd = kwargs.get("ccd", 1)
+    quad = kwargs.get("quad", 1)
+    field = kwargs.get("field", 302)
+    outfile = (
+        os.path.dirname(__file__)
+        + "/../features/field_"
+        + str(field)
+        + "/ccd_"
+        + str(ccd).zfill(2)
+        + "_quad_"
+        + str(quad)
+    )
+
+    limit = kwargs.get("limit", 1000)
+
+    if not hasattr(source_ids, "__iter__"):
+        source_ids = (source_ids,)
+
+    id = 0
+    df_collection = []
+    dmdt_collection = []
+    while 1:
+        skip = id * limit
+        query = {
+            "query_type": "find",
+            "query": {
+                "catalog": features_catalog,
+                "filter": {"_id": {"$in": source_ids}},
+            },
+            "kwargs": {"limit": limit, "skip": skip},
+        }
+        response = kowalski.query(query=query)
+        source_data = response.get("data")
```

Comment:
I have not experimented much with setting the limit, but for limit=1000 for the ZTF_source_features_DR5 catalog, I was able to avoid the timeout issue.